### PR TITLE
Update vssdk

### PR DESCRIPTION
**Pull request by genlu**
Testing this in Roslyn first, will update Arcade once we verified it works.

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -231,7 +231,8 @@
     <SystemThreadingTasksExtensionsVersion>4.5.4</SystemThreadingTasksExtensionsVersion>
     <SQLitePCLRawbundle_greenVersion>2.0.4</SQLitePCLRawbundle_greenVersion>
     <UIAComWrapperVersion>1.1.0.14</UIAComWrapperVersion>
-    <MicrosoftVSSDKBuildToolsVersion>16.8.3036</MicrosoftVSSDKBuildToolsVersion>
+    <MicroBuildPluginsSwixBuildVersion>1.1.33</MicroBuildPluginsSwixBuildVersion>
+    <MicrosoftVSSDKBuildToolsVersion>17.0.39-dev17-g977fccd6</MicrosoftVSSDKBuildToolsVersion>
```
Comment (jasonmalinowski) [inline eng/Versions.props:235]: Does this need to go into a vs-deps branch or does this continue to work in Dev16?

File: NuGet.config [hunk 2]
```diff
@@ -11,6 +11,7 @@
     <add key="vssdk" value="https://pkgs.dev.azure.com/azure-public/vside/_packaging/vssdk/nuget/v3/index.json" />
     <add key="vs-impl" value="https://pkgs.dev.azure.com/azure-public/vside/_packaging/vs-impl/nuget/v3/index.json" />
     <add key="vs-buildservices" value="https://pkgs.dev.azure.com/azure-public/vside/_packaging/vs-buildservices/nuget/v3/index.json" />
+    <add key="MicroBuildToolset" value="https://devdiv.pkgs.visualstudio.com/_packaging/MicroBuildToolset/nuget/v3/index.json" />
```
Comment (jasonmalinowski) [inline NuGet.config:14]: The packages aren't in the vssdk feed?

Comment (genlu) [reply NuGet.config:14]: I'm talking with vseng to make it available in public feed

File: eng/Versions.props (hunk 1 above)
Comment (tmat) [reply eng/Versions.props:235]: This change is only going to dev17 branch.

**Comment by genlu** [comment]
Corresponding change in arcade: https://github.com/dotnet/arcade/pull/7114

**Review by jasonmalinowski — APPROVED** [review]

**Pull request merged**